core/arch/arm/kernel/thread_spmc.c

```diff
@@ -1044,6 +1134,14 @@ static void handle_mem_frag_tx(struct thread_smc_args *args,
 	uint32_t ret_w2 = 0;
 	uint32_t ret_w3 = 0;
 	uint32_t ret_fid = 0;
+	int rc = 0;
+
+	if (IS_ENABLED(CFG_NS_VIRTUALIZATION)) {
+		uint16_t guest_id = endpoint_id >> 16;
+
+		if (!guest_id || virt_set_guest(guest_id))
+			goto out_set_rc;
```

Comment:
At this point rc is still 0, i.e. we will return from the function with FFA_SUCCESS. Is this intentional?

Reply:
Good catch, I'll update.